app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -71,35 +114,73 @@ class PopupView : FrameLayout {
     }
 
     /**
-     * Iterates over all the display fields of the popup to get their values and adds them to the
-     * list of PopupAttributes.
+     * Shows the message in a Snackbar.
      */
-    private fun getPopupAttributeList(): ArrayList<PopupAttribute> {
-        val popupManager = PopupManager(context, popup)
-        val fields: List<PopupField> = popupManager.displayedFields
-        val popupDisplayFieldsWithValues: ArrayList<PopupAttribute> = ArrayList()
-        for (field in fields) {
-            popupDisplayFieldsWithValues.add(
-                PopupAttribute(
-                    field.label,
-                    popupManager.getFormattedValue(field)
-                )
-            )
-        }
-        return popupDisplayFieldsWithValues
+    private fun showSnackbarMessage(message: String) {
+        Snackbar.make(this, message, Snackbar.LENGTH_LONG).show()
     }
 
     /**
-     * Represents an object comprising of Popup display field and its value.
+     * Saves the popup by applying changes to the feature service.
      */
-    data class PopupAttribute(val field: String, val value: String)
+    private fun savePopup() {
+        val editingErrorFuture: ListenableFuture<ArcGISRuntimeException> = popupManager.finishEditingAsync()
+        editingErrorFuture.addDoneListener {
+            try {
+                editingErrorFuture.get()
+
+                val feature: Feature = popupManager.popup.geoElement as Feature
+                val featureTable: FeatureTable = feature.featureTable
+                if (featureTable is ServiceFeatureTable) {
+                    val applyEditsFuture: ListenableFuture<List<FeatureEditResult>> =
+                        featureTable.applyEditsAsync()
+                    applyEditsFuture.addDoneListener {
+                        try {
+                            val featureEditResults: List<FeatureEditResult> = applyEditsFuture.get()
+                            // Check for errors in FeatureEditResults
+                            if (featureEditResults.any { result -> result.hasCompletedWithErrors() }) {
+                                showSnackbarMessage("There was a problem saving the feature")
+                            } else {
+                                showSnackbarMessage("Feature saved successfully")
```

Comment:
As above, need to show also for other types of feature tabes?

Reply:
No, just for ServiceFeatureTable at this time